yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -322,8 +322,10 @@ private[spark] class Client(
     // multiple times, YARN will fail to launch containers for the app with an internal
     // error.
     val distributedUris = new HashSet[String]
-    obtainTokenForHiveMetastore(sparkConf, hadoopConf, credentials)
-    obtainTokenForHBase(sparkConf, hadoopConf, credentials)
+    if (isClusterMode) {
```

Comment:
suggestion from @dougb , but yes, I didn't think that through. It's not for the driver, its for the nodes. Will fix